pkg/certificates/reconciler/certificates.go

```diff
@@ -163,10 +163,13 @@ func (r *reconciler) commitUpdatedSecret(ctx context.Context, secret *corev1.Sec
 	secret = secret.DeepCopy()
```

Comment:
Do you want to set [`secret.Type = "kubernetes.io/tls"` as well](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets)?

Reply:
hmmm I must've missed this spot, I will look into it. 

I was planning on setting the type in the serving code, where (I thought) the secrets were created. That is why I started this one: https://github.com/knative/serving/pull/13388
Though it won't be finished until these protocol changes are in, so that the data gets populated.

Reply:
I opted for changing it in serving because, in my hacking experience on this, Kubernetes doesn't like when you try to change the type of the secret.

Reply:
I realize now I didn't link my serving PR in this description, sorry about that. Just added it in

Reply:
You'd need to delete and re-create the Secret to change the type.

I think that might be worthwhile to do as a one-time switchover, but I'd live to get @eric-sap or @pierDipi or someone else familiar with this code to comment.

Reply:
Oh, but I see lines 76-78 say that we expect the creator of the certificate to set the type, not to set it here.

I'll buy that.